client/src/contexts/UserProvider.js

```diff
@@ -0,0 +1,27 @@
+import React, { createContext, useState, useEffect } from "react";
+const context = createContext(null);
+
+const UserProvider = ({ children }) => {
+    const [user, setUser] = useState({});
+
+    useEffect(() => {
+        fetch("/api/auth/users/me")
+            .then(res => {
+                res.json()
```

Comment:
is this where it is returning `null`?

Reply:
yep

Reply:
Now when I fetch the data from the front-end and check the values of the `console.log(request.session.passport)` and `console.log(request.user)` in the server, I'm seeing the output `undefined for both. I'm guessing the front-end would need to somehow call the back-end auth/google route. I'll try doing that with the signup button.